extensions/binding/org.eclipse.smarthome.binding.yahooweather/src/main/java/org/eclipse/smarthome/binding/yahooweather/handler/YahooWeatherHandler.java

```diff
@@ -127,21 +142,66 @@ public void handleCommand(ChannelUID channelUID, Command command) {
         }
     }
 
+    @Override
+    public Collection<ConfigStatusMessage> getConfigStatus() {
+        Collection<ConfigStatusMessage> configStatus = new ArrayList<>();
+
+        try {
+            String weatherData = getWeatherData();
+            String result = StringUtils.substringBetween(weatherData, "<item><title>", "</title>");
+            if ("City not found".equals(result)) {
+                configStatus.add(ConfigStatusMessage.Builder.error(LOCATION_PARAM, LOCATION_NOT_FOUND)
+                        .withArguments(location).build());
+            }
+        } catch (IOException e) {
+            logger.debug("Communication error occurred while getting Yahoo weather information.", e);
+        }
+
+        return configStatus;
+    }
+
+    @Override
+    public boolean supportsEntity(String entityId) {
+        return getThing().getUID().getAsString().equals(entityId);
+    }
+
+    @Override
+    public void setConfigStatusCallback(ConfigStatusCallback configStatusCallback) {
+        this.configStatusCallback = configStatusCallback;
+    }
+
+    @Override
+    public void handleConfigurationUpdate(Map<String, Object> configurationParameters)
```

Comment:
I don't think so.
If we would ever use Java 8 we could use default implementations. But I never used this until now and I don't think we should do that.

I recently informed Kai that a binding needs to import the config.validation bundle because the BaseThingHandler throws an exception. We could life with this situation or we could try to solve it. As it is an indirect introduced dependency it is IMHO not a good thing. But I don't know how to solve this cleanly.